sphinx_gallery/gen_rst.py

```diff
@@ -231,6 +235,19 @@ def text2string(content):
         return content
 
 
+def extract_thumbnail_number(text):
+    """ Pull out the thumbnail image number specified in the docstring. """
+
+    # check whether the user has specified a specific thumbnail image
+    pattr = re.compile("^\s*#\s*sphinx_gallery_thumbnail_number\s*=\s*([0-9]+)\s*$", flags=re.MULTILINE)
+    match = pattr.search(text)
+    try:
+        thumbnail_number = int(match.groups()[0])
+    except AttributeError:
+        thumbnail_number = 1 # by default, use the first figure created
+
+    return thumbnail_number
```

Comment:
Since we are only using the thumbnail_number and nobody seemed to have any suggestions of other per-example metadata that could be useful, I would be in favour of returning just the thumbnail number for YAGNI reasons.

Maybe worth investigating if it's easy to keep the thumbnail number out of `split_code_and_text_blocks` indeed.